doc/decisions/error_handling.md

```diff
@@ -0,0 +1,39 @@
+# Error Handling
+
+## Problem
+
+Currently, a lot of functions return the error value, regardless of error. These poses some problems, for instance authors of bindings, where it is hard to tell which error exactly occurred. A big improvement to this would be returning different error codes for different types of errors, in order to be able to provide better information in cases of errors. Some functions also make use of an `errorKey`.
+
+While this seems like a decent approach, to provide additional informations in cases of error, it is currently very badly documented. It is not really transparent for the users of the function, what kind of information is included in the `errorKey` and even which messages could possibly be returned.
+
+## Constraints
+
+We also have to consider that many parts of the API very constrained in what values they can use for error codes. If a function returns a pointer the only possible error value by definition is NULL.
+
+## Assumptions
+
+## Considered Alternatives
+
+- Change return types for many functions across the API
+
+## Decision
+
+- Update documentation in `doc/dev/error-*` and link to them in the documentation
+  for the module `kdb`
+- Add second channel for getting information about errors
+- Return error codes directly from functions where failures are expected, e.g. `kdbGet`, `keySetName`
+- Harmonize return values from all functions and move error reporting to second channel
+
+## Rationale
+
+This simplifies and unifies error reporting across all functions. It also introduces
```

Comment:
On the contrary: it would make it more complicated and if we are not very careful less unified.